src/lib/custom-resources/cdk-guardduty-admin-setup/runtime/src/index.ts

```diff
@@ -86,9 +94,10 @@ async function onCreateOrUpdate(
   );
   if (requiredMemberAccounts.length > 0) {
     await createMembers(requiredMemberAccounts, detectorId);
-    await updateMemberDataSource(requiredMemberAccounts, detectorId, s3Protection);
   }
 
+  await updateMemberDataSource(existingMembers, detectorId, s3Protection, eksProtection);
```

Comment:
Moved `updateMemberDataSource` outside the previous if statement and updated to iterate over all existing members. This will enable EKS for existing members. Without this, it will only get enabled on new accounts.